Zscaler/zscaler/block_ioc.py

```diff
@@ -0,0 +1,126 @@
+from collections import defaultdict
+from typing import Dict
+from zscaler.helpers import stix_to_indicators
+
+from sekoia_automation.action import Action
+from zscaler_api_talkers import ZiaTalker
+from requests.exceptions import JSONDecodeError
+
+
+class ZscalerAction(Action):
+    def zia_auth(self):
+        try:
+            api = ZiaTalker(self.module.configuration["base_url"])
+            api.authenticate(
+                api_key=self.module.configuration["api_key"],
+                username=self.module.configuration["username"],
+                password=self.module.configuration["password"],
+            )
+            return api
+        except Exception as e:
+            print(f"ZIA authentication failed: {str(e)}")
+            return None
+
+    def get_valid_indicators_from_list(self, arguments) -> list:
+        try:
+            IOC_list = [arguments["IoC"]]
+            self.log(f"IOC_list to block {IOC_list}")
+            return IOC_list
+        except Exception as e:
+            print(f"Build of IOC list failed: {str(e)}")
+            return []
+
+    def get_valid_indicators_from_stix(self, stix_objects):
+        """
+        Transform the IOC from the STIX format to the payload expected by Zscaler
+        """
+        ZSCALER_IOC_TYPE: Dict[str, Dict[str, str]] = {
+            "ipv4-addr": {"value": "ipv4"},
+            "ipv6-addr": {"value": "ipv6"},
+            "domain-name": {"value": "domain"},
+            "url": {"value": "url"},
+        }
+        seen_values = defaultdict(list)
+        results = {"valid": [], "revoked": []}
+        try:
+            for object in stix_objects:
+                # Extract value and type from pattern
+                print(f"object {object}")
+                indicators = stix_to_indicators(stix_object=object, supported_types_map=ZSCALER_IOC_TYPE)
+                print(f"indicators {indicators}")
```

Comment:
Remove print or replace them by `self.log` if you need to display the information to the customer
```suggestion
                indicators = stix_to_indicators(stix_object=object, supported_types_map=ZSCALER_IOC_TYPE)
```